apps/trip_plan/lib/trip_plan/api/open_trip_planner/parser.ex

```diff
@@ -77,10 +77,14 @@ defmodule TripPlan.Api.OpenTripPlanner.Parser do
   end
 
   defp parse_time(ms_after_epoch) do
-    ms_after_epoch
-    |> DateTime.from_unix!(:millisecond)
-    |> Timex.set(microsecond: {0, 0})
-    |> Timex.to_datetime(OTP.config(:timezone))
+    {:ok, ms_after_epoch_dt} =
+      ms_after_epoch
+      |> DateTime.from_unix!(:millisecond)
+      |> Timex.set(microsecond: {0, 0})
+      |> DateTime.to_unix()
+      |> FastLocalDatetime.unix_to_datetime(OTP.config(:timezone))
```

Comment:
You don't need the extra DataTime conversions here, since you're starting with a Unix epoch time. Sorry for missing this earlier.

```elixir
seconds = Integer.floor_div(ms_after_epoch, 1000)
{:ok, dt} = FastLocalDateTime.unix_to_datretime(seconds, OTP.config(:timezone))
dt
```